web-services/query/src/main/java/datawave/webservice/query/logic/filtered/FilteredQueryLogic.java

```diff
@@ -0,0 +1,383 @@
+package datawave.webservice.query.logic.filtered;
+
+import datawave.audit.SelectorExtractor;
+import datawave.marking.MarkingFunctions;
+import datawave.webservice.common.audit.Auditor;
+import datawave.webservice.common.connection.AccumuloConnectionFactory;
+import datawave.webservice.query.Query;
+import datawave.webservice.query.configuration.GenericQueryConfiguration;
+import datawave.webservice.query.exception.QueryException;
+import datawave.webservice.query.logic.BaseQueryLogic;
+import datawave.webservice.query.logic.QueryLogic;
+import datawave.webservice.query.logic.QueryLogicTransformer;
+import datawave.webservice.query.logic.RoleManager;
+import datawave.webservice.query.result.event.ResponseObjectFactory;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.commons.collections4.iterators.TransformIterator;
+
+import java.security.Principal;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A filtered query logic will only actually execute the delegate query logic if the filter passes. Otherwise this will do nothing and return no results.
+ */
+public class FilteredQueryLogic implements QueryLogic<Object> {
+    
+    private QueryLogic<Object> delegate;
+    private QueryLogicFilter filter;
+    
+    private boolean filtered = false;
+    
+    public FilteredQueryLogic() {}
+    
+    public FilteredQueryLogic(FilteredQueryLogic other) throws CloneNotSupportedException {
+        this.delegate = (QueryLogic<Object>) (other.delegate.clone());
+        this.filter = other.filter;
+        this.filtered = other.filtered;
+    }
+    
+    public QueryLogic<Object> getDelegate() {
+        return delegate;
+    }
+    
+    public void setDelegate(BaseQueryLogic<Object> delegate) {
+        this.delegate = delegate;
+    }
+    
+    public QueryLogicFilter getFilter() {
+        return filter;
+    }
+    
+    public void setFilter(QueryLogicFilter filter) {
+        this.filter = filter;
+    }
+    
+    public interface QueryLogicFilter {
+        boolean canRunQuery(Query settings, Set<Authorizations> auths);
+    }
+    
+    @Override
+    public String getPlan(Connector connection, Query settings, Set<Authorizations> runtimeQueryAuthorizations, boolean expandFields, boolean expandValues)
+                    throws Exception {
+        if (!filtered && filter.canRunQuery(settings, runtimeQueryAuthorizations)) {
+            return delegate.getPlan(connection, settings, runtimeQueryAuthorizations, expandFields, expandValues);
+        } else {
+            filtered = true;
+            return "";
```

Comment:
Wondering if an empty plan is the correct thing to do here. I guess it comes down to the question: Is there some way to record the fact that the query was now run because is was 'filtered' elsewhere in the metrics?

Reply:
If this query logic was filtered out, then there was no plan generated.  Are you suggesting that I should return something like "<filtered>" here?  The lack of plan is a pretty good indication that nothing was run.

Reply:
Good point. I was thinking it would be helpful to know that the query wasn't run because this specific query logic decided that it should not run, might be helpful for folks troubleshooting that aren't as familiar with the system. Nevertheless, it's not clear how valuable that might be in the long run.